src/AppBundle/Controller/Report/DocumentController.php

```diff
@@ -37,4 +34,57 @@ public function add(Request $request, $reportId)
 
         return ['id' => $document->getId()];
     }
+
+    /**
+     * GET document by id
+     *
+     * @Route("/document/{id}")
+     * @Method({"GET"})
+     */
+    public function getOneById(Request $request, $id)
+    {
+        $serialisedGroups = $request->query->has('groups')
+            ? (array) $request->query->get('groups') : ['documents'];
+        $this->setJmsSerialiserGroups($serialisedGroups);
+
+        /* @var $document Document */
+        $document = $this->findEntityBy(Document::class, $id);
+
+        $this->denyAccessIfClientDoesNotBelongToUser($document->getReport()->getClient());
+
+        return $document;
+    }
+
+
+    /**
+     * Delete document.
+     *
+     * @Method({"DELETE"})
+     *
+     * @Route("/document/{id}")
+     *
+     * @param int $id
+     *
+     * @return array
+     */
+    public function delete($id)
+    {
+        $this->get('logger')->debug('Deleting document ' . $id);
+
+        try {
+            /** @var $document Document $note */
+            $document = $this->findEntityBy(Document::class, $id);
+
+            // enable if the check above is removed and the note is available for editing for the whole team
+            $this->denyAccessIfClientDoesNotBelongToUser($document->getReport()->getClient());
+
+            $this->getEntityManager()->remove($document);
+
+            $this->getEntityManager()->flush($document);
+        } catch (\Exception $e) {
+            $this->get('logger')->error('Failed to delete document ID: ' . $id . ' - ' . $e->getMessage());
```

Comment:
exception should be re-thrown in order for the listener to convert it into a readable response for the client